src/ripple/protocol/impl/AccountID.cpp

```diff
@@ -25,9 +25,135 @@
 
 namespace ripple {
 
+#ifndef RIPPLED_ACCOUNTID_CACHE_SIZE
+#define RIPPLED_ACCOUNTID_CACHE_SIZE 1000000
+#endif
+
+namespace detail {
+
+/** Caches the base58 representations of AccountIDs */
+class AccountIdCache
+{
+private:
+    // The total number of items the cache buffer can contain
+    std::size_t const count_;
+
+    // The maximum size (in bytes) of an item, along with a null terminator
+    // and aligned at 8 bytes
+    std::size_t const size_ = 40;
+
+    // The offset of the next available block of `size_` contiguous bytes
+    std::size_t index_ = 0;
+
+    // The memory block
+    std::unique_ptr<char[]> chars_;
+
+    // The map
+    hardened_hash_map<AccountID, char*> map_;
+
+    std::mutex mutable mutex_;
+
+    // The total number of requests to format an AccountID
+    std::size_t requests_ = 0;
+
+    // The number of times were able to satisfy the request from cache
+    std::size_t hits_ = 0;
+
+public:
+    AccountIdCache(std::size_t count)
+        : count_(count)
+        , chars_(std::make_unique<char[]>(count_ * size_))
+        , map_(count_ / 971)
```

Comment:
why `count_ / 971`?

Reply:
I wanted to give the map "reasonable" number of buckets; 971 is a prime number and assuming a reasonable hash function (not an entirely unreasonable assumption, I think), this would mean that, once full, the map would be expected to have _about_ 1,000 items per bucket.

Reply:
My understanding is that the bucket_count provided is just an initial count, and that the bucket_count will increase as the table fills up to prevent the linked list in each bucket to grow too large (lists of 1000 entries / bucket would be very inefficient). I think maybe an initialization with `971` (instead of `count_ / 971`) would be my choice.